packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -120,54 +133,139 @@ export type EditableTreeOrPrimitive = EditableTree | PrimitiveValue;
 /**
  * EditableTree, but with these cases of unwrapping:
  * - primitives are unwrapped. See {@link EditableTreeOrPrimitive}.
- * - nodes with PrimaryField are unwrapped to just the primaryField. See `getPrimaryField`.
+ * - nodes with PrimaryField (see `getPrimaryField`) are unwrapped to {@link EditableField}s.
  * - fields are unwrapped based on their schema's multiplicity. See {@link UnwrappedEditableField}.
  */
-export type UnwrappedEditableTree = EditableTreeOrPrimitive | readonly UnwrappedEditableTree[];
+export type UnwrappedEditableTree = EditableTreeOrPrimitive | EditableField;
 
 /**
- * A field of an {@link EditableTree}.
+ * A field of an {@link EditableTree} as an array-like sequence of unwrapped nodes (see {@link UnwrappedEditableTree}).
+ *
+ * The number of nodes depends on a field's multiplicity.
+ * When iterating, the nodes are read at once. Use index access to read the nodes "lazily".
+ * Use `getWithoutUnwrapping` to get a node without unwrapping.
  */
-export type EditableField = readonly [FieldSchema, FieldKey, readonly EditableTree[]];
+export interface EditableField extends ArrayLike<UnwrappedEditableTree> {
+    /**
+     * The `FieldSchema` of this field.
+     */
+    readonly fieldSchema: FieldSchema;
+
+    /**
+     * The `FieldKey` of this field.
+     */
+    readonly fieldKey: FieldKey;
+
+    /**
+     * Gets the type of the node by its index or the "field type" if index is not provided and this is the primary field.
+     * If the node is well-formed, it must follow its schema.
+     * @param index - if index is provided, returns the type of the node. Otherwise, if this field is the primary field,
+     * returns the type of the parent node this primary field belongs to.
+     * @param nameOnly - if true, returns only the type identifier.
+     */
+    getType(index?: number, nameOnly?: boolean): NamedTreeSchema | TreeSchemaIdentifier | undefined;
```

Comment:
I think we should avoid methods like this, where they do rather different things depending on their arguments. Its better to have multiple methods for the different use cases since it makes the callers much stronger typed, and saves the implementer from having to check which thing its supposed to do.

Reply:
First I thought to refactor this and the node's `getType` in a separate PR, but it seems to be fine to get rid of this for `EditableField` already here. So, removed and replaced with `EditableField.primaryType`.